### PR TITLE
list of basic rich text field options

### LINKED ISSUE #7756: [PNI] - Create new set of options for rich text fields
In products.py, there are a number of rich text fields in which we only give the following options:
` features=['bold', 'italic', 'link']`
instead of defining these for every new field we want to make rich text, lets make this a constant so all we need to do is call on this list, and have all rich text fields have a unified list of options.

### PR DESCRIPTION
**Pull request by danielfmiranda**
Closes #7756 

The purpose of this PR is to have a uniform list of "base" fields for rich text that we can use for the majority of rich text fields in the CMS.

If a user requests a custom list of features, we can either enter a list of those features if they do not include "bold, italic, and link" ( for example the block quotes can only use bold), or, we can append a list of additional features that they want after base_rich_text_fields.

### CODE REVIEW

File: network-api/networkapi/wagtailpages/pagemodels/customblocks/recent_blog_entries.py [hunk 1]
```diff
@@ -3,10 +3,14 @@
...
 
-from ..blog.blog_category import BlogPageCategory
 from networkapi.wagtailpages.utils import get_locale_from_request
 
 
+def get_blog_page_categories():
+    from ..blog.blog_category import BlogPageCategory  # noqa
+    return BlogPageCategory.get_categories
+
+
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/customblocks/recent_blog_entries.py:13]: why would this be required? (the categories don't

File: network-api/networkapi/wagtailpages/pagemodels/base.py [hunk 2]
```diff
@@ -175,9 +176,7 @@ class ParticipatePage2(PrimaryPage):
     )
 
     ctaHeroSubhead = RichTextField(
-        features=[
-            'bold', 'italic', 'link',
-        ],
+        features=(base_rich_text_options),
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/base.py:179]: Putting `()` around things is a dangerous thing in Python - at best it does nothing, but often it completely changes the data type, so let's not add them here =)

File: network-api/networkapi/wagtailpages/pagemodels/blog/blog_category.py [hunk 3]
```diff
@@ -51,16 +51,21 @@ def get_categories():
...
         """
-        choices = [(cat.name, cat.name) for cat in BlogPageCategory.objects.all()]
-        choices.sort(key=lambda c: c[1])
+        choices = []
+        try:
+            choices = [(cat.name, cat.name) for cat in BlogPageCategory.objects.all()]
+            choices.sort(key=lambda c: c[1])
+        except Exception as err:
+            print(type(err))
+            pass
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/blog/blog_category.py:68]:
```suggestion
        except ProgrammingError:
            pass
```

File: network-api/networkapi/wagtailpages/pagemodels/blog/blog_category.py [hunk 4]
```diff
@@ -6,6 +6,8 @@
 from wagtail.core.models import TranslatableMixin
 from wagtail.snippets.models import register_snippet
 
+from networkapi.wagtailpages.pagemodels.customblocks.base_rich_text_options import base_rich_text_options
+
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/blog/blog_category.py:10]:
```suggestion
from django.db.utils import ProgrammingError
```
Although we probably want this directly under the other db import

**Review by Pomax — CHANGES_REQUESTED** [review]

File: network-api/networkapi/wagtailpages/pagemodels/blog/blog_category.py (hunk 3 above)
Comment (danielfmiranda) [reply network-api/networkapi/wagtailpages/pagemodels/blog/blog_category.py]: Added to latest commit! Marking as resolved

File: network-api/networkapi/wagtailpages/pagemodels/blog/blog_category.py (hunk 4 above)
Comment (danielfmiranda) [reply network-api/networkapi/wagtailpages/pagemodels/blog/blog_category.py:11]: Added to latest commit! Marking as resolved

File: network-api/networkapi/wagtailpages/pagemodels/base.py (hunk 2 above)
Comment (danielfmiranda) [reply network-api/networkapi/wagtailpages/pagemodels/base.py]: Added to latest commit! Marking as resolved

File: network-api/networkapi/wagtailpages/pagemodels/blog/blog_category.py (hunk 4 above)
Comment (danielfmiranda) [reply network-api/networkapi/wagtailpages/pagemodels/blog/blog_category.py:11]: Added to latest commit! Marking as resolved

File: network-api/networkapi/wagtailpages/pagemodels/article_fields.py [hunk 5]
```diff
@@ -13,7 +14,7 @@
     )),
     ('card_grid', customblocks.CardGridBlock()),
     ('content', customblocks.ArticleRichText(
-        features=['bold', 'italic', 'h2', 'h3', 'h4', 'h5', 'link', 'large', 'ol', 'ul', 'image', 'hr', 'footnotes'],
+        features=base_rich_text_options + ['h2', 'h3', 'h4', 'h5', 'large', 'ol', 'ul', 'image', 'hr', 'footnotes'],
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/article_fields.py:17]: does it make sense to unify this with the blog's list as one shared set that is `full_content_rich_text_options = base_rich_text_options + ['h2', 'h3', 'h4', 'h5', 'ol', 'ul', 'hr',]` and then have blog use that directly, with article using `full_content_rich_text_options + ['large', 'image', 'footnotes']`?

File: network-api/networkapi/wagtailpages/pagemodels/customblocks/image_text_block.py [hunk 6]
```diff
@@ -1,10 +1,11 @@
 from wagtail.core import blocks
 from .image_block import ImageBlock
+from ..customblocks.base_rich_text_options import base_rich_text_options
 
 
 class ImageTextBlock(ImageBlock):
     text = blocks.RichTextBlock(
-        features=['bold', 'italic', 'h2', 'h3', 'h4', 'h5', 'h6', 'ol', 'ul', 'link']
+        features=base_rich_text_options + ['h2', 'h3', 'h4', 'h5', 'h6', 'ol', 'ul']
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/customblocks/image_text_block.py:8]: I suspect we don't need h6 here at all (no on should ever need h6), so we might be about to unify this with the "Expanded"blog/article list too?

File: network-api/networkapi/wagtailpages/pagemodels/youtube.py [hunk 7]
```diff
@@ -30,11 +31,10 @@ class YoutubeRegretsPage(FoundationMetadataPageMixin, Page):
     faq = StreamField(
         [
             ('paragraph', blocks.RichTextBlock(
-                features=[
-                    'bold', 'italic',
+                features=base_rich_text_options + [
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/youtube.py:34]: this also feels pretty close to blog/article

**Review by Pomax — CHANGES_REQUESTED** [review]

**Comment by danielfmiranda** [comment]
Hi @Pomax, thanks for making some great suggestions regarding a new list with the full functionality of rich text fields. I have added that to this PR and updated the related pages. Thanks again!

File: network-api/networkapi/wagtailpages/pagemodels/customblocks/base_fields.py [hunk 8]
```diff
@@ -9,12 +10,7 @@
 
 base_fields = [
     ('paragraph', blocks.RichTextBlock(
-        features=[
-            'bold', 'italic', 'large',
-            'h2', 'h3', 'h4', 'h5',
-            'ol', 'ul',
-            'link', 'hr',
-        ],
+        features=base_rich_text_options + ['large', 'h2', 'h3', 'h4', 'h5', 'ol', 'ul', 'hr'],
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/customblocks/base_fields.py:13]: Looking at

```
full_content_rich_text_options = base_rich_text_options + ['h2', 'h3', 'h4', 'h5', 'ol', 'ul', 'hr']
```

this looks like it should be `features=full_content_rich_text_options  + ['large']`?

File: network-api/networkapi/wagtailpages/pagemodels/customblocks/recent_blog_entries.py [hunk 9]
```diff
@@ -40,6 +40,8 @@ class RecentBlogEntries(blocks.StructBlock):
     #       error indication what's wrong.
 
     def get_context(self, value, parent_context=None):
+        from ..blog.blog_category import BlogPageCategory  # noqa
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/customblocks/recent_blog_entries.py:43]: This line should not be necessary here, given that it's already on line 7 as well.

File: network-api/networkapi/wagtailpages/pagemodels/blog/blog_category.py [hunk 10]
```diff
@@ -51,16 +52,20 @@ def get_categories():
...
         to apply this function from its original location.
         """
-        choices = [(cat.name, cat.name) for cat in BlogPageCategory.objects.all()]
-        choices.sort(key=lambda c: c[1])
+        choices = []
+        try:
+            choices = [(cat.name, cat.name) for cat in BlogPageCategory.objects.all()]
+            choices.sort(key=lambda c: c[1])
+        except ProgrammingError:
+            pass
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/blog/blog_category.py:68]: Let's add some comments to explain to future readers why this exception is here.

**Review by Pomax — CHANGES_REQUESTED** [review]

File: network-api/networkapi/wagtailpages/pagemodels/customblocks/recent_blog_entries.py [hunk 11]
```diff
@@ -40,6 +40,7 @@ class RecentBlogEntries(blocks.StructBlock):
     #       error indication what's wrong.
 
     def get_context(self, value, parent_context=None):
+
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/pagemodels/customblocks/recent_blog_entries.py:43]:
```suggestion
```

**Review by Pomax — APPROVED** [review]

**Pull request merged**